otri/filtering/filter_list.py

```diff
@@ -1,90 +1,92 @@
 
         Parameters:
             layer : FilterLayer
                 Non-empty layer of filters.
         '''
-        self.layers.append(layer)
+        self.__layers.append(layer)
 
-    def execute(self, on_atom_output: Callable = None, on_execute_finished: Callable = None):
+    def execute(self, source: Mapping[str, Stream], on_data_output: Callable = None):
         '''
-        Starts working on the origin stream with the given filter layers
+        Works on the source streams with the given filter layers
 
         Parameter:
-            source_streams : Collection[Streams]
-                Atoms collections. Required iterator that has `has_next()` method.
+            source : Mapping[str : Streams]
+                Source streams.
+            on_data_output : Callable
+                Function called everytime a filter from the last layer outputs something in any of its output layers.
         '''
-        # Grab the output streams
-        last_output_streams = self.layers[len(
-            self.layers) - 1][0].get_output_streams()
-        last_output_iterators = [x.__iter__() for x in last_output_streams]
+        self.stream_dict.update(source)
+        # Setup phase
+        for filter_layer in self.__layers:
+            for f in filter_layer:
+                f.setup(self.__get_streams_by_names(f.get_inputs()),self.__get_streams_by_names(f.get_outputs()), self.status_dict)
+
+        # Execute phase
+        while(not self.__is_all_finished()):
+            for filter_layer in self.__layers:
+                for fil in filter_layer:
+                    fil.execute()
+        return self
 
-        self.__execute_0(last_output_iterators, on_atom_output)
+    def streams(self) -> Mapping[str, Stream]:
+        return self.stream_dict
 
-        if (on_execute_finished != None):
-            on_execute_finished(self)
+    def status(self, key: str, default : Any) -> Any:
+        return self.status_dict.get(key,default)
 
     def __is_all_finished(self) -> bool:
         '''
-        Checks if the last filter layer's filters are flagged as finished
+        Checks if the last filter layer's filters' output streams are flagged as closed.
+        All streams must be initialised inside the self.stream_dict class variable.
         '''
-        if not self.layers[len(self.layers) - 1].is_finished():
-            return False
-        return True
 
-    def __execute_0(self, last_output_iterators, on_atom_output):
-        while(not self.__is_all_finished()):
-            for filter_layer in self.layers:
-                for fil in filter_layer:
-                    fil.execute()
-            # If any filter has outputted any atom, call the on_atom_output method
-            if on_atom_output != None:
-                for iterator in last_output_iterators:
-                    while(iterator.has_next()):
-                        on_atom_output(next(iterator))
+        for filter in self.__layers[len(self.__layers) - 1]:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'filter'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)